### PR TITLE
unify electrum forks

### PR DESCRIPTION
**Pull request by prusnak**
###### Description of changes

- supersededes https://github.com/NixOS/nixpkgs/pull/207752 + contains more cleanups

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/electrum/grs.nix [hunk 1]
```diff
@@ -63,8 +63,14 @@ python3.pkgs.buildPythonApplication {
...
   ];
 
-  preBuild = ''
-    substituteInPlace ./electrum_grs/ecc_fast.py \
+  postPatch = ''
+    # make compatible with protobuf4 by easing dependencies ...
+    substituteInPlace ./contrib/requirements/requirements.txt \
+      --replace "protobuf>=3.12,<4" "protobuf>=3.12"
+    # ... and regenerating the paymentrequest_pb2.py file
+    protoc --python_out=. electrum/paymentrequest.proto
```
Comment (gruve-p) [inline pkgs/applications/misc/electrum/grs.nix:71]:
```suggestion
    protoc --python_out=. electrum_grs/paymentrequest.proto
```

**Comment by prusnak** [comment]
I see the same build error as @lourkeur here: https://github.com/NixOS/nixpkgs/pull/207752#issuecomment-1365108178

@gruve-p please have a look and advise

File: pkgs/applications/misc/electrum/grs.nix [hunk 2]
```diff
@@ -63,8 +63,14 @@ python3.pkgs.buildPythonApplication {
...
-  preBuild = ''
-    substituteInPlace ./electrum_grs/ecc_fast.py \
+  postPatch = ''
+    # make compatible with protobuf4 by easing dependencies ...
+    substituteInPlace ./contrib/requirements/requirements.txt \
+      --replace "protobuf>=3.12,<4" "protobuf>=3.12"
+    # ... and regenerating the paymentrequest_pb2.py file
+    protoc --python_out=. electrum_grs/paymentrequest.proto
+
+   substituteInPlace ./electrum_grs/ecc_fast.py \
```
Comment (gruve-p) [inline pkgs/applications/misc/electrum/grs.nix:73]:
```suggestion
    substituteInPlace ./electrum_grs/ecc_fast.py \
```

**Comment by gruve-p** [comment]
> I see the same build error as @lourkeur here: [#207752 (comment)](https://github.com/NixOS/nixpkgs/pull/207752#issuecomment-1365108178)
> 
> @gruve-p please have a look and advise

I am able to replicate however I do not know why the build is hanging

**Comment by prusnak** [comment]
> I am able to replicate however I do not know why the build is hanging

I disabled the tests for electrum-grs completely - let's fix in subsequent PRs.

File: pkgs/applications/misc/electrum/grs.nix (hunk 2 above)
Comment (prusnak) [reply pkgs/applications/misc/electrum/grs.nix]: applied in 7c1eeef919affdd564a01ef0631891d5428eb8f2

**Comment by prusnak** [comment]
@GrahamcOfBorg build electrum electrum-ltc electrum-grs

**Review by bbjubjub2494 — APPROVED** [review]
Result of `nixpkgs-review pr 207834` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>electrum-grs</li>
  </ul>
</details>

`electrum-grs` appears to be functional.

**Pull request merged**